src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,125 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double INITIAL_LOAD_FACTOR = 0.75;
+    private Entry<K, V>[] table = new Entry[INITIAL_CAPACITY];
+    private int size = 0;
+
     @Override
     public void put(K key, V value) {
-
+        if (size >= INITIAL_LOAD_FACTOR * table.length) {
+            resizing();
+        }
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int hashCode = key.hashCode();
+        int index = (table.length - 1) & hashCode;
```

Comment:
Create informative variable and method names. Writing proper variable names can be a highly valuable investment to the quality of your code. Not only you and your teammates understand your code better, but it can also improve code sustainability in the future later on. When you go back to the same code base and re-read it over again, you should understand what is going on.

Reply:
It is informative!